Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contests/contest-details/ContestDetailsPage.tsx

```diff
@@ -0,0 +1,165 @@
+import { useNavigate } from 'react-router';
+import { Link, useParams } from 'react-router-dom';
+
+import ContestBreadcrumbs from '../../../components/contests/contest-breadcrumbs/ContestBreadcrumbs';
+import Button, { ButtonSize, ButtonState } from '../../../components/guidelines/buttons/Button';
+import Heading, { HeadingType } from '../../../components/guidelines/headings/Heading';
+import SpinningLoader from '../../../components/guidelines/spinning-loader/SpinningLoader';
+import useTheme from '../../../hooks/use-theme';
+import { setContestDetails } from '../../../redux/features/contestsSlice';
+import { useGetContestByIdQuery } from '../../../redux/services/contestsService';
+import { useAppDispatch, useAppSelector } from '../../../redux/store';
+import { flexCenterObjectStyles } from '../../../utils/object-utils';
+
+import styles from './ContestDetailsPage.module.scss';
+
+const ContestDetailsPage = () => {
+    const dispatch = useAppDispatch();
+    const navigate = useNavigate();
+    const { contestId } = useParams();
+    const { isLoggedIn } = useAppSelector((state) => state.authorization);
+    const { themeColors, getColorClassName } = useTheme();
+    const { contestDetails, selectedCategory } = useAppSelector((state) => state.contests);
+    const { data, isLoading, error } = useGetContestByIdQuery({ id: Number(contestId) });
+
+    const textColorClassName = getColorClassName(themeColors.textColor);
+    const {
+        id,
+        name,
+        allowedSubmissionTypes,
+        description,
+        problems,
+        competeParticipantsCount,
+        practiceParticipantsCount,
+        canBeCompeted,
+        canBePracticed,
+    } = data ?? {};
+
+    useEffect(() => {
+        if (contestDetails?.id !== data?.id) {
+            dispatch(setContestDetails({ contest: data ?? null }));
+        }
+    }, [ data, contestDetails, dispatch ]);
+
+    const renderAllowedLanguages = () => allowedSubmissionTypes?.map((allowedSubmissionType) => (
+        <>
+            <Link
+              key={`contest-sub-strat-btn-${allowedSubmissionType.id}`}
+              className={styles.allowedLanguageLink}
+              to={`/contests?category=${selectedCategory?.id}&strategy=${allowedSubmissionType.id}`}
+            >
+                {allowedSubmissionType.name}
+            </Link>
+            {' | '}
+        </>
+    ));
+
+    const renderProblemsNames = () => {
+        if (!problems || problems.length === 0) {
+            return 'The problems for this contest are not public.';
+        }
+        return problems.map((problem) => (
+            <div className={styles.problemNameItem}>
+                {problem.name}
+            </div>
+        ));
+    };
+
+    const renderContestActionButton = (isCompete: boolean) => {
+        const navigateToUrl = isCompete
+            ? `/contests/${id}/compete`
+            : `/contests/${id}/practice`;
+        const isDisabled = isCompete
```

Comment:
The contests route is passed to many places , it will be better to extract constant